### DISCUSSION TITLE
Ensure the completions from some backend have the highest priority

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [company-mode/company-mode#1144] [Q&A] [answered]
See the following completion results for my case and the corresponding company-backends settings:

![image](https://user-images.githubusercontent.com/11155854/126465220-dee2bf94-8a7a-4e2a-ba21-9ce3f3d0e839.png)

I want to ensure that the completion candidates returned from `company-tabnine` have the highest priority. How to achieve this?

Regards,
HY

**Comment by dgutov** [comment]
What your value of `company-transformers`?

**Comment by hongyi-zhao** [comment]
See the screenshot below:

![image](https://user-images.githubusercontent.com/11155854/126503043-32a1d26d-e4db-4958-8267-144c38298032.png)

**Comment by dgutov** [comment, marked as answer]
* Using `:separate` in the backends group is right: it will keep completions separate.
* `company-transformers` should be either empty, or carefully constructed not to mess with the order across backends.

Seeing as you have `company-prescient-transformer` in there, either try removing it or see whether `company-prescient` has any relevant configuration.

**Comment by hongyi-zhao** [comment]
Got it. Thank you very much. 

According to the documentation [here](https://github.com/raxod502/prescient.el#company-specific), customizing the user option `company-prescient-sort-length-enable` to `nil` will fix this problem:

```emacs-lisp
(use-package prescient :demand t)

(use-package company-prescient
  :after prescient
  :init (setq company-prescient-sort-length-enable nil)
  :config (company-prescient-mode t))
```
![image](https://user-images.githubusercontent.com/11155854/126573764-be732722-40b0-4c38-a2ef-97417b4c4d94.png)

I also tried to comment out all the configuration related to the `company-prescient` package, which can achieve the same result:

![image](https://user-images.githubusercontent.com/11155854/126573258-c0baf174-c575-4d18-87a2-a71191edcefe.png)

**Comment by dgutov** [comment]
I'll try converting this to a Discussion (Q&A). Let's see whether it will convert all the answers into one thread.

**Reply by dgutov** [reply]
It did not. :-(

**Reply by hongyi-zhao** [reply]
I really don't know what's the prerequisite for combining all the answers into one thread via converting an issue to a Discussion (Q&A). Do you mean this method can be used to combine all issues similar in content and initiated by the same author into one?

**Reply by dgutov** [reply]
Probably not.

This was an experiment, we got what we got.

**Reply by yugaego** [reply]
> It did not. :-(

That's a bummer. I don't see any documentation for such cases. So seems a safe transfer would be possible only for issues with maximum one comment.

**Reply by dgutov** [reply]
Or at least those that have One Important Comment. The rest can be deleted or hidden, as I did here.

They could provide some alternative ideas too, though.

**Reply by yugaego** [reply]
A more correct statement from my side could have been: Seems a safe transfer would be possible only for issues with one answering comment from one author.

> The rest can be deleted or hidden, as I did here.

The result is almost very good with those hidden secondary comments: the two important answers are prominently visible.
Except for the "5 suggested answers" label and having collapsed comments, the rest seems to be what was desired.

> They could provide some alternative ideas too

Additionally, an author potentially might be not very happy about one's comment removal, so I'd say at maximum hidden.
